File: src/main/java/com/wezaam/withdrawal/service/UserService.java
package com.wezaam.withdrawal.service;

import com.wezaam.withdrawal.model.User;

import java.util.List;
import java.util.Optional;

public interface UserService {

    User addUser(User user);
    void deleteUser(Long userId);
    Optional<User> getUser(Long userId);
    List<User> findAllUsers();
    User findByName(String name) throws Exception;
}

File: src/main/java/com/wezaam/withdrawal/service/UserService.kt
package com.wezaam.withdrawal.service

import com.wezaam.withdrawal.model.User
import com.wezaam.withdrawal.repository.UserRepository
import org.springframework.stereotype.Service

@Service
class UserService(private val userRepository: UserRepository) {
    fun findAll(): List<User?>? = userRepository.findAll()
    fun findById(id: Long) = userRepository.findById(id)
}